src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/JndiDirectoryDAO.scala

```diff
@@ -255,9 +285,21 @@ class JndiDirectoryDAO(protected val directoryConfig: DirectoryConfig)(implicit
     groups.toSet
   }
 
-  override def enableUser(userId: WorkbenchUserId): Future[Unit] = withContext { ctx =>
+  override def enableUser(userId: WorkbenchUserId): Future[Unit] = {
```

Comment:
I think this function is all biz logic now and belongs in UserService. Make enableIdentity public then when setting up the pet SA just enable that identity.

Reply:
Gotcha - will do